Horion/Module/Modules/Spider.cpp

```diff
@@ -0,0 +1,45 @@
+#include "Spider.h"
+
+Spider::Spider() : IModule('Z', Category::MOVEMENT, "Climb walls") {
+	registerFloatSetting("Speed", &speed, speed, 0.1, 1);
+}
+
+Spider::~Spider() {
+}
+
+const char* Spider::getModuleName() {
+	return "Spider";
+}
+
+void Spider::onMove(C_MoveInputHandler* input) {
+	if (g_Data.getLocalPlayer() == nullptr)
+		return;
+	C_LocalPlayer* player = g_Data.getLocalPlayer();
+
+	if (player->isInLava() == 1 || player->isInWater() == 1)
+		return;
+
+	if (player->isSneaking())
+		return;
+
+	if (!(input->forward || input->backward) || (input->forward && input->backward))
+		return;
+	
+	static std::vector<vec3_ti*> sideBlocks;
+	if (sideBlocks.empty()) {
+		sideBlocks.push_back(new vec3_ti(0, 0, -1));
+		sideBlocks.push_back(new vec3_ti(0, 0, 1));
+		sideBlocks.push_back(new vec3_ti(-1, 0, 0));
+		sideBlocks.push_back(new vec3_ti(1, 0, 0));
+	}
+
+	for (auto current : sideBlocks) {
```

Comment:
a nested for-loop would have been more appropriate here imo

Reply:
if im not mistaken there even is a boolean inside the player struct that is true when the player collides with walls horizontally

Reply:
Not sure what property you're talking about... I tried to look for it before coming up with the side-block check